src/compiler.ts

```diff
@@ -8004,13 +8004,13 @@ export class Compiler extends DiagnosticEmitter {
       );
     }
     assert(index == numArgumentsInclThis);
-    return this.makeCallIndirect(signature, indexArg, reportNode, operands, immediatelyDropped);
+    return this.makeCallIndirect(signature, functionPointerExpr, reportNode, operands, immediatelyDropped);
   }
 
-  /** Creates an indirect call to the function at `indexArg` in the function table. */
+  /** Creates an indirect call to the function at `functionInstance` in the function table. */
```

Comment:
The comment here needs to be updated now as well. Perhaps

```suggestion
  /** Creates an indirect call to the first-class function pointed to by `functionPointerExpr`. */
```